### PR TITLE
upgrade to org.apache.commons.fileupload2

### PR DESCRIPTION
**Pull request by lukamalovic**
This branch contains the necessary changes to namespaces, method calls, maven dependency names and versions, etc. to upgrade from the commons.fileupload API to version 2. My ability to verify these changes was limited by the commons-chain library causing the maven build to fail.

### CODE REVIEW

File: core/src/main/java/org/apache/struts/upload/CommonsMultipartRequestHandler.java [hunk 1]
```diff
@@ -443,7 +446,7 @@ protected void addTextParameter(HttpServletRequest request, FileItem item) {
 
         if (!haveValue) {
             try {
-                value = item.getString("ISO-8859-1");
+                value = item.getString(Charset.forName("ISO-8859-1"));
```
Comment (ste-gr) [inline core/src/main/java/org/apache/struts/upload/CommonsMultipartRequestHandler.java:449]: Here you could use `StandardCharsets.ISO_8859_1`

**Comment by ste-gr** [comment]
Hi Luka,

I wouldn't change the "old" release-notes, since the `commons-fileupload` was the current version at the time. Maybe just update the link itself to https://github.com/apache/commons-fileupload/tree/1.x

Greetings
Stefan

**Comment by lukamalovic** [comment]
Just made the changes as per your suggestions, let me know if you have any further questions/concerns!

**Comment by ste-gr** [comment]
Hi Luka,

I added a new branch [weblegacy_chain](https://github.com/weblegacy/struts1/tree/weblegacy_chain). Maybe you could use it to make a functional SNAPSHOT-Version.

Greetings
Stefan

**Pull request merged**